numba/np/random/distributions.py

```diff
@@ -574,3 +574,164 @@ def random_logseries(bitgen, p):
             return 1
         else:
             return 2
+
+
+@register_jitable
+def random_binomial_btpe(bitgen, n, p):
+    r = min(p, 1.0 - p)
+    q = 1.0 - r
+    fm = n * r + r
+    m = np.floor(fm)
+    p1 = np.floor(2.195 * np.sqrt(n * r * q) - 4.6 * q) + 0.5
+    xm = m + 0.5
+    xl = xm - p1
+    xr = xm + p1
+    c = 0.134 + 20.5 / (15.3 + m)
+    a = (fm - xl) / (fm - xl * r)
+    laml = a * (1.0 + a / 2.0)
+    a = (xr - fm) / (xr * q)
+    lamr = a * (1.0 + a / 2.0)
+    p2 = p1 * (1.0 + 2.0 * c)
+    p3 = p2 + c / laml
+    p4 = p3 + c / lamr
+
+    case = 0
+    y = k = 0
+    while 1:
+        if case == 0:
+            nrq = n * r * q
+            u = next_double(bitgen) * p4
+            v = next_double(bitgen)
+            if (u > p1):
+                case = 1
+                continue
+            y = np.floor(xm - p1 * v + u)
+            case = 6
+            continue
+        elif case == 1:
+            if (u > p2):
+                case = 2
+                continue
+            x = xl + (u - p1) / c
+            v = v * c + 1.0 - abs(m - x + 0.5) / p1
+            if (v > 1.0):
+                case = 0
+                continue
+            y = np.floor(x)
+            case = 4
+            continue
+        elif case == 2:
+            if (u > p3):
+                case = 3
+                continue
+            y = np.floor(xl + np.log(v) / laml)
+            if ((y < 0) or (v == 0.0)):
+                case = 0
+                continue
+            v = v * (u - p2) * laml
+            case = 4
+            continue
+        elif case == 3:
+            y = np.floor(xr - np.log(v) / lamr)
+            if ((y > n) or (v == 0.0)):
+                case = 0
+                continue
+            v = v * (u - p3) * lamr
+            case = 4
+            continue
+        elif case == 4:
+            k = abs(y - m)
+            if ((k > 20) and (k < ((nrq) / 2.0 - 1))):
+                case = 5
+                continue
+            s = r / q
+            a = s * (n + 1)
+            F = 1.0
+            if (m < y):
+                for i in range(m + 1, y + 1):
+                    F = F * (a / i - s)
+            elif (m > y):
+                for i in range(m + 1, y + 1):
```

Comment:
I think this range expression is incorrect compared to NumPy (it's backwards). Consider the `if..else`, the range domain must reflect the nature of what is in the predicates.

Reply:
Resolved in [3dfb3ca](https://github.com/numba/numba/pull/8371/commits/3dfb3cad5a2629fe52672ad71534e215ee3063e1)